### PR TITLE
feat: use configured pk type in Action Mailbox migration

### PR DESCRIPTION
**Pull request by nshki**
### Motivation / Background

I recently ran Action Mailbox's generators on a new Rails 7 project and noticed that it didn't carry over my generator primary key configuration. I had to manually modify the migration in my project. I know that Active Storage is a gem that uses a project's configured primary key, so I thought this would be a great opportunity for Action Mailbox to do the same.

### Detail

This PR adds a `migrations_test.rb` for Action Mailbox that closely mirrors the same file in Active Storage. It then adds a private `primary_key_type` method in the Action Mailbox migration which pulls a project's configured primary key type. This method is also closely modeled after Active Storage's.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: actionmailbox/db/migrate/20180917164000_create_action_mailbox_tables.rb [hunk 1]
```diff
@@ -10,4 +10,11 @@ def change
       t.index [ :message_id, :message_checksum ], name: "index_action_mailbox_inbound_emails_uniqueness", unique: true
     end
   end
+
+  private
+    def primary_key_type
+      config = Rails.configuration.generators
+      setting = config.options[config.orm][:primary_key_type]
+      setting || :primary_key
```
Comment (eileencodes) [inline actionmailbox/db/migrate/20180917164000_create_action_mailbox_tables.rb:18]: I'd prefer to copy the exact code from ASt for this https://github.com/rails/rails/blob/main/activestorage/db/update_migrate/20191206030411_create_active_storage_variant_records.rb#L16-L19

File: actionmailbox/test/dummy/db/migrate/20180208205311_create_action_mailbox_tables.rb [hunk 2]
```diff
@@ -10,4 +10,11 @@ def change
       t.index [ :message_id, :message_checksum ], name: "index_action_mailbox_inbound_emails_uniqueness", unique: true
     end
   end
+
+  private
+    def primary_key_type
+      config = Rails.configuration.generators
+      setting = config.options[config.orm][:primary_key_type]
+      setting || :primary_key
+    end
```
Comment (eileencodes) [inline actionmailbox/test/dummy/db/migrate/20180208205311_create_action_mailbox_tables.rb:19]: Same comment as the other `primary_key_type`.

File: actionmailbox/test/migrations_test.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      CreateActionMailboxTables.migrate(:down)
+      CreateActionMailboxTables.migrate(:up)
+    end
+
+    def action_mailbox_tables
+      [:action_mailbox_inbound_emails]
+    end
+
+    def primary_key(table)
+      puts "TEEHEE"
```
Comment (eileencodes) [inline actionmailbox/test/migrations_test.rb:50]:
```suggestion
```

File: actionmailbox/test/migrations_test.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      puts @connection.columns(table).inspect
```
Comment (eileencodes) [inline actionmailbox/test/migrations_test.rb:51]:
```suggestion
```

**Review by eileencodes — CHANGES_REQUESTED** [review]
Hi @nshki thanks for the PR! I've left some comment for changes that need to be made. Once you've made those changes please squash your commits into one. We generally don't merge multi-commit PRs that are related to a single change.

File: actionmailbox/test/migrations_test.rb (hunk 3 above)
Comment (nshki) [reply actionmailbox/test/migrations_test.rb:50]: Ha, this is embarrassing. Apologies for not taking out my debug comments!

Comment (eileencodes) [reply actionmailbox/test/migrations_test.rb]: No worries, we've all done it 😄

**Comment by nshki** [comment]
Thanks for the review @eileencodes! I went ahead and addressed your comments and squashed the commits into one. 🙇🏻 I also renamed this PR title to be a little shorter so it's friendlier as a commit message.

**Pull request merged**